### PR TITLE
LPS-135136 Create set method for account type via headless API

### PR DESCRIPTION
**Pull request by erickmonteiroo**
Waiting for release.

### CODE REVIEW

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by Sharryxue** [comment]
The failure in [rest-builder-jdk8/0/0](https://testray.liferay.com/zh/home/-/testray/case_results/1545992952?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D1545992736%26testrayRunId%3D1545992766%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc) is more like related, hello @erickmonteiroo, would you please take a further look? Thanks

**Comment by erickmonteiroo** [comment]
ci:test:relevant

File: modules/apps/account/account-rest-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -46,6 +46,12 @@ components:
                     type: integer
                 status:
                     type: integer
+                type:
+                    # @review
+                    description:
+                        The account's type. This property will be ignored during PATCH and PUT requests. Valid types are
+                        'business', 'guest' and 'person'.
+                    type: string
```
Comment (drewbrokke) [inline modules/apps/account/account-rest-impl/rest-openapi.yaml:54]: I think it makes sense to make this an `enum`.

File: modules/apps/account/account-rest-impl/src/main/java/com/liferay/account/rest/internal/resource/v1_0/AccountResourceImpl.java [hunk 2]
```diff
@@ -182,8 +182,7 @@ public Account postAccount(Account account) throws Exception {
 		AccountEntry accountEntry = _accountEntryLocalService.addAccountEntry(
 			contextUser.getUserId(), _getParentAccountId(account),
 			account.getName(), account.getDescription(), _getDomains(account),
-			null, null, null, AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS,
-			_getStatus(account), null);
+			null, null, null, account.getType(), _getStatus(account), null);
```
Comment (drewbrokke) [inline modules/apps/account/account-rest-impl/src/main/java/com/liferay/account/rest/internal/resource/v1_0/AccountResourceImpl.java:185]: What happens when you omit the `type` from the POST request? We _should_ still be defaulting to type "business".

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@erickmonteiroo @brunoquei2105 this looks pretty good, just a couple comments from me.

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
Rebased on commerce migration. No conflicts found.

**Comment by drewbrokke** [comment]
@erickmonteiroo now that the migration pull is merged, will you please rebase and resend these changes? The `account-rest-*` modules no longer exist, so you'll need to apply them inside of `headless-admin-user-*`.

**Pull request closed without merging**